docs/nnx/bridge_guide.md

```diff
@@ -0,0 +1,544 @@
+bridge.lazy_init(model, x)                            # => `var = model.init(key, x)` in Linen
+y = model(x)                                          # => `y = model.apply(var, x)` in Linen
+
+print(model.params['w'].shape)  # => (32, 64) - weights are stored within the NNX module
+print(y.shape)                  # => (4, 64)
+
+# In-place swap your weight array and the model still works!
+model.params['w'].value = jax.random.normal(jax.random.key(1), (32, 64))
+assert not jnp.allclose(y, model(x))
+```
+
+    (32, 64)
+    (4, 64)
+
+
+`nnx.bridge.lazy_init` also works even if the top-level module is a pure-NNX one, so you can do sub-moduling as you wish:
+
+
+```python
+class NNXOuter(nnx.Module):
+  def __init__(self, out_dim: int, rngs: nnx.Rngs):
+    self.dot = nnx.bridge.ToNNX(LinenDot(out_dim), rngs=rngs)
+    self.b = nnx.Param(jax.random.uniform(rngs.params(), (1, out_dim,)))
+
+  def __call__(self, x):
+    return self.dot(x) + self.b
+
+x = jax.random.normal(jax.random.key(42), (4, 32))
+model = bridge.lazy_init(NNXOuter(64, rngs=nnx.Rngs(0)), x)  # Can fit them into one line too
+w, b = model.dot.params['w'], model.b
+print(w.shape, b.shape)
+y = model(x)
+print(y.shape)
+```
+
+    (32, 64) (1, 64)
+    (4, 64)
+
+
+The Linen weight is already converted to a typical NNX variable, which is a thin wrapper of the actual JAX array value within. Here, `w` is an `nnx.Param` because it belongs to the `params` collection of `LinenDot` module.
+
+We will talk more about different collections and types in the [NNX Variable <-> Linen Collections](#variable-types-vs-collections) section. Right now, just know that they are converted to NNX variables like native ones.
+
+
+```python
+assert isinstance(model.dot.params['w'], nnx.Param)
+assert isinstance(model.dot.params['w'].value, jax.Array)
+```
+
+### NNX -> Linen
+
+To convert an NNX module to Linen, you should forward your creation arguments to `bridge.ToLinen` and let it handle the actual creation process. 
+
+We don't let you create an NNX module of your own because new variables are created eagerly as soon as a NNX module is created. Instead, `bridge.ToLinen` will handle it for you.
+
+An NNX module instance initializes all its variables eagerly when it is created, which consumes memory and compute. On the other hand, Linen modules are stateless, and the typical `init` and `apply` process involves multiple creation of them. To bypass this issue, you should send your creation arguments (instead of a created NNX module) to `bridge.to_linen`, and let it handle the rest.
+
+
+```python
+class NNXDot(nnx.Module):
+  def __init__(self, in_dim: int, out_dim: int, rngs: nnx.Rngs):
+    self.w = nnx.Param(nnx.initializers.lecun_normal()(rngs.params(), (in_dim, out_dim)))
+  def __call__(self, x: jax.Array):
+    return x @ self.w
+
+x = jax.random.normal(jax.random.key(42), (4, 32))
+model = bridge.ToLinen(NNXDot, args=(32, 64))  # <- Pass in the arguments, not an actual module
```

Comment:
We should use `to_linen` as its both more ergonomic and does the `FrozenDict` conversion automatically. Most users will copy the first example they see.

Reply:
Done.